arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -710,6 +710,12 @@ public fun <A> Iterable<A>.combineAll(MA: Monoid<A>): A = MA.run {
   }
 }
 
+public fun <A> Iterable<A>.combineOrNull(SG: Semigroup<A>): A? = SG.run {
```

Comment:
Thanks @Quantum64 for your first contribution!

Awesome that you've followed naming of the Kotlin Std! 
Should we name it `combineAllOrNull` to be in line with the `combineAll` methods that relies on `Monoid`?

(CI is red due to some issues on Kotlin MPP & Github Actions cross-builds. We're working on it)

Reply:
I agree that `combineAllOrNull` is a more appropriate name